Applications/Spire/Include/Spire/Ui/AssociativeValueModel.hpp

```diff
@@ -1,38 +1,47 @@
 #ifndef SPIRE_ASSOCIATIVE_VALUE_MODEL_HPP
 #define SPIRE_ASSOCIATIVE_VALUE_MODEL_HPP
 #include <functional>
+#include <boost/functional/hash.hpp>
 #include <boost/signals2/shared_connection_block.hpp>
 #include "Spire/Ui/Ui.hpp"
 #include "Spire/Ui/LocalValueModel.hpp"
 
-namespace Spire {
-namespace Details {
-  struct AssociatedBooleanModel {
-    std::shared_ptr<BooleanModel> m_model;
-    boost::signals2::scoped_connection m_connection;
-  };
-
-  template <typename T, typename U = boost::optional<T>>
-  struct OptionalHash {
-    std::size_t operator()(const U& value) const;
+namespace std {
+  template<typename T>
+  struct hash<boost::optional<T>> {
+    size_t operator ()(const boost::optional<T>& value) const {
+      if(!value) {
```

Comment:
Remove this check.

Reply:
Use ```boost::hash_value```.

Reply:
Boost doesn't provide a hash function for boost::optional. It has one for std::optional, though.